src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -228,6 +232,14 @@ protected ExecutionStrategy(DataFetcherExceptionHandler dataFetcherExceptionHand
 
         GraphqlFieldVisibility fieldVisibility = executionContext.getGraphQLSchema().getFieldVisibility();
         Map<String, Object> argumentValues = valuesResolver.getArgumentValues(fieldVisibility, fieldDef.getArguments(), field.getArguments(), executionContext.getVariables());
+        Map<String, Map<String, Object>> directiveArgumentValues = field.getDirectives().stream()
+                .collect(Collectors.toMap(Directive::getName, dir -> {
+                    GraphQLDirective directive = executionContext.getGraphQLSchema().getDirective(dir.getName());
+                    if (directive == null) {
+                        return Collections.emptyMap();
+                    }
+                    return valuesResolver.getArgumentValues(fieldVisibility, directive.getArguments(), dir.getArguments(), executionContext.getVariables());
+                }));
 
```

Comment:
Can we make put this into a DirectivesUtil helper class so its simpler to read and easier to test

    Map<String, Map<String, Object>> directiveArgumentValues = DirectivesUtil.readDirectiveValues(field)

See `graphql.DirectivesUtil`

Reply:
I think this should be a Map<String,graphql.schema.GraphQLDirective>

That is a field has a map of named `graphql.schema.GraphQLDirective` objects.  From that you can get the arguments and the values to this arguments.

We have other code somewhere (off hand) that helps create that